src/lib/storage/disk_analyzer.rb

```diff
@@ -0,0 +1,368 @@
+      # /control.xml in the inst-sys. We cannot simply compare device
+      # minor/major IDs since the inst-sys is in a RAM disk (copied from the
+      # installation medium).
+      #
+      def installation_volume?(vol)
+        log.info("Checking if #{vol.name} is an installation volume")
+        is_inst = mount_and_check(vol) { |mp| installation_volume_check(mp) }
+        log.info("#{vol} is installation medium") if is_inst
+        is_inst
+      end
+
+      # Check if the volume mounted at 'mount_point' is an installation volume.
+      # This is a separate method so it can be redefined in unit tests.
+      #
+      # @return 'true' if it is an installation volume, 'false' if not.
+      #
+      def installation_volume_check(mount_point)
+        check_file = "/control.xml"
+        mount_point += "/" unless mount_point.end_with?("/")
+        return false unless File.exist?(mount_point + check_file)
+        FileUtils.identical?(check_file, mount_point + check_file)
+      end
+
+      # Mount a volume, perform the check given in 'block' while mounted, and
+      # then unmount. The block will get the mount point of the volume as a
+      # parameter.
+      #
+      # @return the return value of 'block' or 'nil' if there was an error.
+      #
+      def mount_and_check(vol, &block)
+        raise ArgumentError, "Code block required" unless block_given?
+        mount_point = "/mnt" # FIXME
+        begin
+          # check if we have a filesystem
+          # return false unless vol.filesystem
+          mount(vol.name, mount_point)
+          check_result = block.call(mount_point)
+          umount(mount_point)
+          check_result
+        rescue RuntimeError => ex
+          log.error("CAUGHT exception: #{ex} for #{vol}")
+          nil
+        end
+      end
+
+      # Return an array of the device names of the specified block devices
+      # (::Storage::Disk, ::Storage::Partition, ...).
+      #
+      # @param disks [Array<BlkDev>]
+      # @return [Array<string>] names
+      #
+      def dev_names(blk_devices)
+        blk_devices.map { |x| x.to_s }
+      end
+
+      # Mount a device.
+      #
+      # This is a temporary workaround until the new libstorage can handle that.
+      #
+      def mount(device, mount_point)
+        # FIXME: use libstorage function when available
+        cmd = "/usr/bin/mount #{device} #{mount_point} >/dev/null 2>&1"
+        log.info("Trying to mount #{device}: #{cmd}")
+        raise "mount failed for #{device}" unless system(cmd)
+      end
+
+      # Unmount a device.
+      #
+      # This is a temporary workaround until the new libstorage can handle that.
+      #
+      def umount(mount_point)
+        # FIXME: use libstorage function when available
+        cmd = "/usr/bin/umount #{mount_point}"
+        log.info("Unmounting: #{cmd}")
+        raise "umount failed for #{mount_point}" unless system(cmd)
+      end
```

Comment:
That's what the mount_and_check method in line #319 does. It just uses those two methods as helpers.